### PR TITLE
Add google cloud endpoints to handle the files in the storage

### PR DESCRIPTION
**Pull request by avalmas-programize**
GET /google-cloud/ : Fetch dangling files
DELETE /google-cloud/ : Delete dangling files

### CODE REVIEW

File: packages/api/src/google-cloud/google-cloud.service.ts [hunk 1]
```diff
@@ -19,6 +25,45 @@ export class GoogleCloudService {
...
+      throw new InternalServerErrorException();
+    }
+
+    const surveyMedia = await this.surveyMediaRepository.find({
+      select: ['url'],
+    });
+
+    const mediaSet = new Set(
+      surveyMedia.map(
+        (media) => media.url.split(`${process.env.GCS_BUCKET}/`)[1],
```
Comment (ericboucher) [inline packages/api/src/google-cloud/google-cloud.service.ts:42]: let's create a util to get the file name out of the GCS url

File: packages/api/src/google-cloud/google-cloud.service.ts [hunk 2]
```diff
@@ -19,6 +25,45 @@ export class GoogleCloudService {
     });
   }
 
+  public async findDanglingFiles(): Promise<string[]> {
+    if (!process.env.GCS_BUCKET || !process.env.STORAGE_FOLDER) {
```
Comment (ericboucher) [inline packages/api/src/google-cloud/google-cloud.service.ts:29]: In general, I prefer using process.env at the very top of the file only to create the constants that you will need. This makes it a bit clearer what depends on external configs

**Comment by ericboucher** [comment]
Looks good, just a few minor comments

File: packages/api/src/google-cloud/google-cloud.service.ts (hunk 1 above)
Comment (avalmas-programize) [reply packages/api/src/google-cloud/google-cloud.service.ts]: Done

File: packages/api/src/google-cloud/google-cloud.service.ts (hunk 2 above)
Comment (avalmas-programize) [reply packages/api/src/google-cloud/google-cloud.service.ts]: Done

File: packages/api/src/google-cloud/google-cloud.service.ts [hunk 3]
```diff
@@ -3,30 +3,79 @@ import {
...
+import { SurveyMedia } from '../surveys/survey-media.entity';
+import { getFileFromURL } from '../utils/google-cloud.utils';
 
 @Injectable()
 export class GoogleCloudService {
   private logger = new Logger('GCS');
   private storage: Storage;
 
-  constructor() {
+  private readonly GCS_BUCKET = process.env.GCS_BUCKET;
```
Comment (ericboucher) [inline packages/api/src/google-cloud/google-cloud.service.ts:17]: Is it possible to use destructuring here maybe?

**Review by ericboucher — APPROVED** [review]
One more minor comment but looks good, you can merge after

File: packages/api/src/google-cloud/google-cloud.service.ts (hunk 3 above)
Comment (avalmas-programize) [reply packages/api/src/google-cloud/google-cloud.service.ts:17]: No not inside a class.

**Pull request merged**